src/metabase/db/custom_migrations.clj

```diff
@@ -277,21 +288,38 @@
                                  _ [[k v]]))
                              column_settings)))))))
 
-(define-migration AddJoinAliasToVisualizationSettingsFieldRefs
+(define-reversible-migration AddJoinAliasToColumnSettingsFieldRefs
+  (let [update-one! (fn [{:keys [id visualization_settings] :as card}]
+                  (let [updated (add-join-alias-to-column-settings-refs card)]
+                    (when (not= visualization_settings updated)
+                      (t2/query-one {:update :report_card
+                                     :set    {:visualization_settings updated}
+                                     :where  [:= :id id]}))))]
+    (run! update-one! (t2/reducible-query {:select [:id :visualization_settings :result_metadata]
+                                           :from   [:report_card]
+                                           :where  [:and
+                                                    [:or
+                                                     [:= :query_type nil]
+                                                     [:= :query_type "query"]]
+                                                    [:like :visualization_settings "%\\\\\"ref\\\\\",[\\\\\"field%"]
+                                                    [:like :result_metadata "%\"join-alias\"%"]]})))
```

Comment:
I added these to the where clause as performance optimizations, which will dramatically speed up the migration. Line 304 will filter for cards where the users have customized a column with a field ref like `[:field ...]`, and the line 305 will filter for cards with fields from joins. The tests still pass so we're guaranteed this works